### DISCUSSION TITLE
Show modal when back browser

### DISCUSSION THREAD

**Discussion by mfaridzia** [vercel/next.js#29047] [Help] [answered]
How to show modal when user back browser?

if i use alert from js, it works, but when i want to display modal component it doesn't work.

```
useEffect(() => {
    window.history.pushState(null, '', document.URL);
    window.addEventListener('popstate', () => {
        alert("Modal!")
    });
}, [])
```

**Comment by wilsonneto-swe** [comment]
Why aren't you using the next router? I think will be better, this way will prevent you from breaking changes...

```
const router = useRouter();
...
router.beforePopState(() => setPopupVisible(true))
...
...
<Popup visible={popupVisible} />
...
`````

I hope it helps you.

**Reply by mfaridzia** [reply]
Thanks, but I have tried it, if using alert works, but modal component does not work.
so I'm a little confused.

**Reply by wilsonneto-swe** [reply]
How are using the modal? Could you show some code?

**Reply by mfaridzia** [reply]
Okay, here's the code:

```
const [showModal, setShowModal] = useState(false);

useEffect(() => {
    router.beforePopState(() => {
        setShowModal(true);
        return true;
    });
}, [router]);

// inside jsx
{ showModal && <ModalComponent /> }
```

**Comment by wilsonneto-swe** [comment, marked as answer]
@mfaridzia look at this example code, please.

I built a example, you can see it running here:
https://nextjs-back-interception-example-oc7hghy41-wilsonneto-dev.vercel.app/

Navigate to "other page", then try to return with back button, see if this is what you need.

The repository with the code is here:
https://github.com/wilsonneto-dev/nextjs-back-interception-example

I hope it helps you.

**Reply by mfaridzia** [reply]
@wilsonneto-dev  ok I got it, thank you.
